src/components/_banner.scss

```diff
@@ -0,0 +1,99 @@
+@use "../design-tokens/color";
+@use "../design-tokens/typography";
+
+@mixin Manifold__Banner {
+  position: relative;
+  display: flex;
+  box-sizing: border-box;
+  width: 100%;
+  padding: 10px 16px 10px;
+  background: color.$color-white;
+  border-color: color.$color-utilitiesGray;
+  border-style: solid solid solid none;
+  border-width: 1px;
+  border-radius: 4px;
+
+  &::before {
+    position: absolute;
+    top: -1px;
+    left: 0;
+    width: 3px;
+    height: 100%;
+    background-color: color.$color-utilitiesGray;
+    border: 1px solid color.$color-utilitiesGray;
+    border-radius: 4px 0 0 4px;
+    content: '';
+  }
+
+  &__Icon svg {
+    fill: color.$color-blue;
+  }
+
+  p {
+    flex-grow: 1;
+    flex-shrink: 1;
+    margin-top: 1px;
+    margin-bottom: 0;
+    @include typography.Manifold__Typography__SmallScreenBody;
+    @media (min-width: 600px) {
+      @include typography.Manifold__Typography__Body;
+    }
+  }
+
+  &--Error {
+    &::before {
+      background-color: color.$color-red;
+      border: 1px solid color.$color-red;
+    }
+
+    .Manifold__Banner__Icon svg {
+      fill: color.$color-red;
+    }
+  }
+}
+
+@mixin Manifold__Banner--Error {
+  &::before {
+    background-color: color.$color-red;
+    border: 1px solid color.$color-red;
+  }
+
+  .Manifold__Banner__Icon svg {
```

Comment:
All of this looks great! I agree with the decisions you made, and I think this is good to merge.

This isn’t a request for change, but your work has got me thinking: could we improve this long selector here `.Manifold__Banner--Error .Manifold__Banner__Icon svg`?

Having multiple CSS classes chained together starts the cascade wars because it’s difficult to overwrite. Lead down the road to `!important`, it does. But here… maybe it’s OK because we want this icon to be “locked-in.”

But that also begs the question: if we want this icon to be locked-in… why not just make it part of the CSS (as a background-image)? If we want the specific icon & color to be locked in, we could do that and save people from having to manage it. E.g.:

```scss
@mixin Manifold__Banner--Error {
  background-image: url('data/image+svg;base64…');
  padding-left: 1.5rem;
```

Or, alternatively, what if we _do_ want people to be able to change the icon, but not the color? If that’s what we want, maybe it’d be best to just only use:

```scss
@mixin Manifold__Banner--Error {
  > svg {
    fill: mercury.$color-red;
```

And then treat any `<svg>` at the top-level as special (and drop the `__Icon`) mixin in the process:

```html
<div role="alert" class="Manifold__Banner">
  <svg … /> <!-- our special buddy -->
  <p><!-- normal text --></p>
</div>
```

This saves people from having to manage CSS classes & mixins. Does it break the single-depth pattern? Yeah. But it does so because we _want_ that control here.

---

Which of the above 2 options make more sense?

Reply:
Thanks for the thoughtful review @dangodev!

I'm of the opinion that option 2 – setting the svg color via a top-level declaration – is the better option here. I'd like to keep the icons changeable but with sensible defaults. Locking in an icon wouldn't allow for an implementation where we would want a specific icon on a banner that was intending to be more specific – for instance a Plan Builder feature icon could be used instead of a flag if it was a warning specific to that Plan Builder feature. 

It's a tough one though because the elegance of option 1 is very appealing, especially with the lack of HTML (specifically the svg element) that's needed.

Reply:
Makes sense!